### PR TITLE
Optimize convolution operations

### PR DESCRIPTION
**Pull request by inwchun**
optimize convolution operation

### CODE REVIEW

File: src/smt.cpp [hunk 1]
```diff
@@ -119,8 +119,12 @@ vector<Expr> from1DIdx(
...
   for (size_t ii = dims.size(); ii > 0; --ii) {
     size_t i = ii - 1;
-    idxs.emplace_back(i == 0 ? idx1d : idx1d.urem(dims[i]));
-    idx1d = idx1d.udiv(dims[i]);
+    uint64_t v;
+    auto isOne = dims[i].isUInt(v) && v == 1;
+
+    // how to make Expr(0)?
+    idxs.emplace_back(i == 0 ? idx1d : (isOne ? Expr::mkBV(0, 32U) : idx1d.urem(dims[i])));
+    idx1d = isOne ? idx1d : idx1d.udiv(dims[i]);
```
Comment (aqjune) [inline src/smt.cpp:127]: This is not necessary because Expr::udiv already does the optimization

File: src/smt.cpp [hunk 2]
```diff
@@ -119,8 +119,12 @@ vector<Expr> from1DIdx(
   // Start from the lowest dimension
   for (size_t ii = dims.size(); ii > 0; --ii) {
     size_t i = ii - 1;
-    idxs.emplace_back(i == 0 ? idx1d : idx1d.urem(dims[i]));
-    idx1d = idx1d.udiv(dims[i]);
+    uint64_t v;
+    auto isOne = dims[i].isUInt(v) && v == 1;
+
+    // how to make Expr(0)?
+    idxs.emplace_back(i == 0 ? idx1d : (isOne ? Expr::mkBV(0, 32U) : idx1d.urem(dims[i])));
```
Comment (aqjune) [inline src/smt.cpp:126]: You can use `Expr::mkBV(0, dims[0])` to use `dims[0]`'s sort.
BTW, does this improve performance?

File: src/smt.cpp (hunk 1 above)
Comment (inwchun) [reply src/smt.cpp:127]: I think the Expr::urem should be optimized? x%1==0

File: src/smt.cpp (hunk 2 above)
Comment (inwchun) [reply src/smt.cpp:126]: printed expr and found (urem var #0001). isn't this 0?

**Pull request merged**